src/process.ts

```diff
@@ -83,7 +85,11 @@ export interface ProcessCallbackDecorator {
  * CreateProcess factory interface
  */
 export interface CreateProcess {
-	(commands: (Command[] | Command)[], callback?: ProcessCallback): Process;
+	<T>(commands: (Command<T>[] | Command<T>)[], callback?: ProcessCallback<T>): Process<T>;
+}
+
+export function createCommandFactory<T>(store?: Store<T>): CommandFactory<T> {
```

Comment:
do you think we should pass the store to the `createCommandFactory`? Feels like that it would be a pattern we wouldn't want to encourage as commands are purposely decoupled from knowing anything about the store? The user would still be able to provide the `T` generic to explicitly define the state type.

Reply:
That's fine. I put it in there as just a way to avoid having to provide the generic twice, but it makes sense that we are trying to decouple the two.